### PR TITLE
Add an extra bit of documentation to the 'AddressDerivation' module

### PR DESCRIPTION
**Pull request by KtorZ**
# Issue Number

#21

# Overview

- [ ] I have added a bit more of documentation to the `AddressDerivation` module (cf the haddock generated doc).

# Comments

### CODE REVIEW

File: src/Cardano/Wallet/AddressDerivation.hs [hunk 1]
```diff
@@ -309,3 +310,51 @@ keyToAddress (Key xpub) =
     Address $ CBOR.toStrictByteString $ encodeAddress xpub encodeAttributes
   where
     encodeAttributes = CBOR.encodeMapLen 0
+
+-- $use
+-- 'Key' and 'Index' allow for reprenting public keys, private keys, hardened
```
Comment (rvl) [inline src/Cardano/Wallet/AddressDerivation.hs]:
```suggestion
-- 'Key' and 'Index' allow for representing public keys, private keys, hardened
```

File: src/Cardano/Wallet/AddressDerivation.hs [hunk 2]
```diff
@@ -309,3 +310,51 @@ keyToAddress (Key xpub) =
...
+-- indexes and soft indexes for various level in a non-ambiguous manner. Those
```
Comment (rvl) [inline src/Cardano/Wallet/AddressDerivation.hs]: Would it be worth clarifying "soft" → "soft (non-hardened)" ?

**Review by rvl — APPROVED** [review]
This is a nice addition.

File: src/Cardano/Wallet/AddressDerivation.hs (hunk 2 above)
Comment (KtorZ) [reply src/Cardano/Wallet/AddressDerivation.hs]: Nice idea :+1:

**Pull request merged**